src/simulator/world.py

```diff
@@ -146,12 +147,12 @@ def update_visualization_json(self):
 
     def dump_all_shapes_info(self):
         # dump visualization info for shapes to the output json file
-        shapes_info_dict = {"shapes": {}}
+        shapes_info_dict = {"shapes" : {}}
         for oid in self.objects:
             if self.objects[oid].shape is not None:
-                # TODO: to be replaced with proper handling of compound objects (if cond)
                 shapes_info_dict["shapes"][oid] = self.objects[oid].dump_shape_info()
-
+                continue
```

Comment:
I guess this is equivalent to an "else". right? It's much more readable with an else.